api/v1beta1/hostedcluster_types.go

```diff
@@ -148,6 +148,13 @@ type HostedClusterSpec struct {
 	// +optional
 	ClusterID string `json:"clusterID,omitempty"`
 
+	// channel is an identifier for explicitly requesting that a non-default
+	// set of updates be applied to this cluster. The default channel will be
+	// contain stable updates that are appropriate for production clusters.
+	//
+	// +optional
+	Channel string `json:"channel,omitempty"`
```

Comment:
are we possibly planning on expanding the fields that add intent for upgrades recommendation? have you considered having .Channel in its own potentially additively growing struct?

Reply:
Personally, I'd have just used `version configv1.ClusterVersionSpec` in `HostedClusterSpec`, and deprecated `Release`, `ClusterID`, and `Channel`.  But since `Channel` is the only thing I _needed_ on the `spec` side here, I punted that sort of pivot to maybe-some-day future work.  I'm certainly happy to pivot this pull request to that approach if it sounds appealing to you.